src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3572,5 +3572,68 @@ public bool TryGetSimilarName(DName name, FieldNameKind fieldNameKind, out strin
             return similar != null &&
                    comparer.Distance(similar) < (name.Value.Length / 3) + 3;
         }
+
+        /// <summary>
+        /// Try to union all table child types and checks if any coercion is necessary. Meant to be called from within table type check loop.
+        /// </summary>
+        /// <param name="argType">Current table child type.</param>
+        /// <param name="arg">Child node.</param>
+        /// <param name="errors">Error container from caller.</param>
+        /// <param name="returnType">Composed new type.</param>
+        /// <param name="needCoercion">Checks if child node needs to be coerced.</param>
+        /// <returns></returns>
+        public bool TryUnionWithCoerce(DType argType, TexlNode arg, IErrorContainer errors, out DType returnType, out bool needCoercion)
```

Comment:
>TryUnionWithCoerce [](http://example.com/codeflow?start=20&length=18)

Does this function need to be recursive to handle nested tables?

Reply:
`AggregateCoercesTo` handles the recursion.